s3/replication/manager/src/s3replicationmanager/app.py

```diff
@@ -17,11 +17,176 @@
+
+HOST = '127.0.0.1'
+PORT = 8080
+
+# Dictionary holding job_id and fdmi record
+#  e.g. jobs = {'jobA': {'K1': 'V1'}}
+jobs = {}
+
+# List of active subscriber
+subscribers = []
+
+# List of jobs in progress
+jobs_inprogress = {}
+
+# Route table declaration
+routes = web.RouteTableDef()
+
+@routes.get('/jobs')
+async def list_jobs(request):
+    """List active jobs
+
+    Handler api to handle multiple query
+
+    """
+    url = request.path_qs
+    url_ob = urlparse(url)
+    query = parse_qs(url_ob.query)
+
+    # Return in progress jobs
+    if 'inProgress' in query:
+        LOG.debug('InProgress query param: {}'.format(query['inProgress']))
+        return web.json_response({'inProgress': jobs_inprogress})
+    # Return job counts
+    elif 'count' in query:
+        return web.json_response({'Number of jobs present': len(jobs)})
+    # Return requested jobs
+    elif 'prefetch' in query:
+        req_list1 = query['prefetch']
+        pf_cnt = req_list1[0]
+        req_list2 = query['subscriber_id']
+        sub_id = req_list2[0]
+        LOG.debug('sub_id is : {}'.format(sub_id))
+        LOG.debug('subscribers are : {}'.format(subscribers))
+
+        if sub_id in subscribers:
+            LOG.debug('jobs keys : {}'.format(list(jobs.keys())))
+            LOG.debug('jobs in progress : {}'.format(jobs_inprogress))
+            if int(pf_cnt) <= len(jobs):
+                job_ids = jobs.keys()
+                for key in job_ids[:pf_cnt]:
+                    add_inprogress = dict([(k, v) for k, v in jobs.items()])
+                    LOG.debug('add inprogress jobs : {}'.format(add_inprogress))
+                    jobs_inprogress.update(add_inprogress)
+                    LOG.debug('in progress jobs : {}'.format(jobs_inprogress))
+                    response = jobs_inprogress.keys()
+                    LOG.debug('inprogress jobs : {}'.format(add_inprogress))
+            else:
+                jobs_inprogress.update(jobs)
+                LOG.debug('in progress jobs : {}'.format(jobs_inprogress))
+                jobs.clear()
+                response = jobs_inprogress.keys()
+            return web.json_response({'Response': response})
+        else:
+            return web.json_response({'Response': 'INVALID subscriber'})
+
+    else:
+        return web.json_response({'jobs': list(jobs.keys())})
+
+@routes.get('/jobs/{job_id}')
+async def get_job(request):
+    """Get job attribute
+
+    Handler api to fetch job attributes
+
+    """
+    id = request.match_info['job_id']
+
+    # Get job's attribute
+    if id in jobs.keys():
+        return web.json_response(jobs[id])
+    else:
+        return web.json_response({'Response': 'ERROR : Job is not present!'})
+
+@routes.post('/jobs')
+async def add_job(request):
+    """Add jobs
+
+    Handler to add jobs to the job queue
+
+    """
+    entries = await request.json()
+    # Add job to the list
+    jobs.update(entries)
+    LOG.debug('Job list : {}'.format(jobs))
```

Comment:
consider printing only count, since again printing all jobs will be too verbose

Reply:
done